ext/pgsql/pgsql.c

```diff
@@ -2147,6 +2148,17 @@ PHP_FUNCTION(pg_trace)
 	}
 	php_stream_auto_cleanup(stream);
 	PQtrace(pgsql, fp);
+	if (trace_mode > 0) {
+#ifdef PQTRACE_REGRESS_MODE
+		if (trace_mode > (PQTRACE_SUPPRESS_TIMESTAMPS|PQTRACE_REGRESS_MODE)) {
+			php_error_docref(NULL, E_WARNING, "Trace mode is invalid");
```

Comment:
Please throw a ValueError for new code as this is a programming error.

Reply:
ah I always forget, thx.